### DISCUSSION TITLE
flet python does not work

### DISCUSSION THREAD

**Discussion by tarovq** [flet-dev/flet#3362] [Q&A]
import flet as ft
import sqlite3

def fetch_data(table_name):
    connection = sqlite3.connect('post_office.db')
    cursor = connection.cursor()
    cursor.execute(f"SELECT * FROM {table_name}")
    rows = cursor.fetchall()
    cursor.execute(f"PRAGMA table_info({table_name})")
    columns_info = cursor.fetchall()
    column_names = [col[1] for col in columns_info]
    connection.close()
    return column_names, rows

def add_customer(initials, adress, phone, email):
    connection = sqlite3.connect('post_office.db')
    cursor = connection.cursor()
    cursor.execute("INSERT INTO Customers (Initials, Adress, Phone, Email) VALUES (?, ?, ?, ?)",
                   (initials, adress, phone, email))
    connection.commit()
    connection.close()

def main(page: ft.Page):
    page.title = "Post Office Management System"
    page.bgcolor = '#ebebeb'
    page.window_maximized = True

    initials = ft.TextField(label="Initials")
    adress = ft.TextField(label="Adress")
    phone = ft.TextField(label="Phone")
    email = ft.TextField(label="Email")
    add_customer_button = ft.Button(text="Add Customer", on_click=lambda e: add_customer(
        initials.value, adress.value, phone.value, email.value))

    page.add(initials, adress, phone, email, add_customer_button)

    page.run()

ft.app(target=main)
when I run the code, nothing happens and so with any code where there is import flet as ft

**Comment by ndonkoHenri** [comment]
Remove `page.run()`, and rename `Button` to one of the available buttons (see [docs](https://flet.dev/docs/controls/buttons)).
![image](https://github.com/flet-dev/flet/assets/98978078/b6a15990-18ec-47a3-bb4e-0786c4f75068)

Make sure to check your terminal/console, as you will receive error messages when you try to make use of stuffs not present in Flet.

**Reply by tarovq** [reply]
I fixed everything, nothing is called anyway, even if I write any codes with flet, even the simplest ones, nothing happens, the code is executed but the application is not called
![image](https://github.com/flet-dev/flet/assets/170893785/b852d753-cfe0-4524-a109-d78758efa0ff)

**Comment by ndonkoHenri** [comment]
Can you share more details to your environment (OS, Flet version etc)?
I suggest you create a new venv, then install flet and retry

**Comment by leodpj** [comment]
import flet as ft
from flet import ElevatedButton
import sqlite3

#import ElevatedButton and change ft.Button to ElevatedButton as shown


    add_customer_button = ElevatedButton(text="Add Customer", on_click=lambda e: add_customer(
        initials.value, adress.value, phone.value, email.value))

    page.add(initials, adress, phone, email, add_customer_button)


ft.app(target=main)


<img width="353" alt="flet" src="https://github.com/flet-dev/flet/assets/52355982/b4aa51d1-30e8-45d5-be71-4a716c5552f5">


💯
